presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestAddExchangesPlans.java

```diff
@@ -95,4 +132,42 @@ public void testRepartitionForUnionAllBeforeHashJoin()
                                                 anyTree(
                                                         tableScan("region", ImmutableMap.of("regionkey", "regionkey"))))))));
     }
+
+    @Test
+    public void testNonSpillableBroadcastJoinAboveTableScan()
+    {
+        assertDistributedPlan(
+                "SELECT * FROM nation n join region r on n.nationkey = r.regionkey",
+                spillEnabled(BROADCAST),
```

Comment:
It looks like you're enabling spill for broadcast joins.
inline (or rename, if you can find a good name)